### PR TITLE
Unit Test - Test CreateGroup View Model

### PR DESCRIPTION
**Pull request by at-haole2**
#### Related issues (Required)
- Task #194 
- Bug #

#### Description (Required)
 - Init Test CreateGroup View Model

### CODE REVIEW

File: app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private lateinit var groupRepository: GroupRepository
+
+    private lateinit var viewModel: CreateGroupViewModel
+
+    @Before
+    fun initTests() {
+        MockitoAnnotations.initMocks(this)
+        viewModel = CreateGroupViewModel(groupRepository)
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt]: remove this line

File: app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Test
+    fun `Given name and userId - When get name and userId - Then return right`() {
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt]: change this line to `fun Given valid groupName and valid userId - When create group - Then return true() {`

File: app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        /* When */
+        `when`(groupRepository.createGroup(name, userId)).thenReturn(Single.just(true))
+
+        /* Then */
+        viewModel.createGroup(name, userId).subscribe(test)
+        test.assertValue { it == true }
+    }
+
+    @Test
+    fun `Given name and userId - When get name and userId - Then return wrong`() {
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt]: change this line to 
`fun Given invalid groupName and invalid userId - When create group - Then return false() {`

File: app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        val userId = ""
+
+        /* When */
+        `when`(groupRepository.createGroup(name, userId)).thenReturn(Single.just(false))
+
+        /* Then */
+        viewModel.createGroup(name, userId).subscribe(test)
+        test.assertValue { it == false }
+    }
+}
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt]: add break line
and create a 2 case below:
`valid groupName and invalid userId`
`invalid groupName and valid userId`

**Comment by at-haole2** [comment]
I fixed comments. Please check again. Thanks!

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Test
+    fun `Given valid groupName and valid userId - When create group - Then return true`() {
+        /* Given */
+        val groupName = "name"
+        val userId = "userid"
+        val test = TestObserver<Boolean>()
+
+        /* When */
+        `when`(groupRepository.createGroup(groupName, userId)).thenReturn(Single.just(true))
```
Comment (at-tienhoang) [inline app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt]: This is given

File: app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+        /* Then */
+        viewModel.createGroup(groupName, userId).subscribe(test)
```
Comment (at-tienhoang) [inline app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt]: This is `when`

File: app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        val groupName = ""
+        val userId = "user 01"
+        val test = TestObserver<Boolean>()
+
+        /* When */
+        `when`(groupRepository.createGroup(groupName, userId)).thenReturn(Single.just(false))
+
+        /* Then */
+        viewModel.createGroup(groupName, userId).subscribe(test)
+        test.assertValue { !it }
```
Comment (at-tienhoang) [inline app/src/test/java/vn/asiantech/way/CreateGroupViewModelTest.kt]: Should `it == false`

**Comment by at-haole2** [comment]
a @at-tienhoang 
I fixed comments. Please check again. Thanks!

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**